client/example_test.go

```diff
@@ -0,0 +1,279 @@
+package client_test
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal"
+	"github.com/google/go-tpm-tools/server"
+	"github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm/tpm2"
+)
+
+var tpmHashAlg = tpm2.AlgSHA256
+var hashAlg = crypto.SHA256
+
+func ExampleKey_Quote() {
+	// On verifier, make the nonce.
+	nonce := make([]byte, 8)
+
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		log.Fatalf("failed to create nonce: %v", err)
+	}
+
+	// On client machine, generate the TPM quote.
+	// TODO: use real TPM.
+	simulator, err := simulator.Get()
+	if err != nil {
+		log.Fatalf("failed to initialize simulator: %v", err)
+	}
+	defer simulator.Close()
+
+	ak, err := client.AttestationKeyECC(simulator)
+	if err != nil {
+		log.Fatalf("failed to create attestation key: %v", err)
+	}
+	defer ak.Close()
+
+	pcr7 := tpm2.PCRSelection{
+		Hash: tpm2.AlgSHA256,
+		PCRs: []int{7},
+	}
+
+	quote, err := ak.Quote(pcr7, nonce)
+	if err != nil {
+		log.Fatalf("failed to create quote: %v", err)
+	}
+
+	// On verifier, verify the quote against a stored public key/AK
+	// certificate's public part and the nonce passed.
+	if err := internal.VerifyQuote(quote, ak.PublicKey(), nonce); err != nil {
+		// TODO: handle verify error.
+		log.Fatalf("failed to verify quote: %v", err)
+	}
+	// Output:
```

Comment:
I also added `// Output:` comments (https://blog.golang.org/examples) where it made sense. Basically, in examples not related to the event log.

Reply:
Great, now those examples can with go test.